alfresco-transform-tika/alfresco-transform-tika/src/main/java/org/alfresco/transformer/tika/parsers/ExifToolParser.java

```diff
@@ -0,0 +1,260 @@
+
+    public ExifToolParser() throws IOException, TikaException {
+        super();
+        ExternalParser eParser = ExternalParsersFactory.create(getExternalParserConfigURL()).get(0);
+        this.setCommand(eParser.getCommand());
+        this.setIgnoredLineConsumer(eParser.getIgnoredLineConsumer());
+        this.setMetadataExtractionPatterns(eParser.getMetadataExtractionPatterns());
+        this.setSupportedTypes(eParser.getSupportedTypes());
+    }
+    
+    private URL getExternalParserConfigURL(){
+        ClassLoader classLoader = ExifToolParser.class.getClassLoader();
+        return classLoader.getResource(EXIFTOOL_PARSER_CONFIG);
+    }
+
+    /**
+     * Adapted from {@link org.apache.tika.parser.external.ExternalParser} 
+     * due to errors attempting to {@link #extractMetadata} from the errorStream in original implementation.  <p>
+     * Executes the configured external command and passes the given document
+     *  stream as a simple XHTML document to the given SAX content handler.
+     * Metadata is only extracted if {@link #setMetadataExtractionPatterns(Map)}
+     *  has been called to set patterns.
+     */
+    public void parse(InputStream stream, ContentHandler handler, Metadata metadata, ParseContext context)
+            throws IOException, SAXException, TikaException {
+        XHTMLContentHandler xhtml = new XHTMLContentHandler(handler, metadata);
+
+        TemporaryResources tmp = new TemporaryResources();
+        try {
+            parse(TikaInputStream.get(stream, tmp), xhtml, metadata, tmp);
+        } finally {
+            tmp.dispose();
+        }
+    }
+
+    private void parse(TikaInputStream stream, XHTMLContentHandler xhtml, Metadata metadata, TemporaryResources tmp)
+            throws IOException, SAXException, TikaException {
+        boolean inputToStdIn = true;
+        boolean outputFromStdOut = true;
+        boolean hasPatterns = (getMetadataExtractionPatterns() != null && !getMetadataExtractionPatterns().isEmpty());
+
+        File output = null;
+
+        // Build our getCommand()
+        String[] cmd;
+        if (getCommand().length == 1) {
+            cmd = getCommand()[0].split(" ");
+        } else {
+            cmd = new String[getCommand().length];
+            System.arraycopy(getCommand(), 0, cmd, 0, getCommand().length);
+        }
+        for (int i = 0; i < cmd.length; i++) {
+            if (cmd[i].indexOf(INPUT_FILE_TOKEN) != -1) {
+                cmd[i] = cmd[i].replace(INPUT_FILE_TOKEN, stream.getFile().getPath());
+                inputToStdIn = false;
+            }
+            if (cmd[i].indexOf(OUTPUT_FILE_TOKEN) != -1) {
+                output = tmp.createTemporaryFile();
+                outputFromStdOut = false;
+                cmd[i] = cmd[i].replace(OUTPUT_FILE_TOKEN, output.getPath());
+            }
+        }
+
+        // Execute
+        Process process = null;

```

Comment:
Add DEBUG (or TRACE ?) to log the command ?

Reply:
If the issue we currently have is fixed by Tika, we should remove this whole class and just use the standard functionality of `ExternalParser`. This only modification I have made to these methods is to remove a line of code that was producing the bug (which was at approx L146: `extractMetadata(err, metadata);`
I don't think we should be adding additional logging now, with expectation of removing this class in the future.